pkg/ccl/streamingccl/streamingest/stream_ingestion_processor.go

```diff
@@ -385,42 +456,33 @@ func (sip *streamIngestionProcessor) Next() (rowenc.EncDatumRow, *execinfrapb.Pr
 		return nil, sip.DrainHelper()
 	}
 
-	sip.mu.Lock()
-	err := sip.mu.pollingErr
-	sip.mu.Unlock()
-	if err != nil {
-		sip.MoveToDraining(err)
-		return nil, sip.DrainHelper()
-	}
-
-	progressUpdate, err := sip.consumeEvents()
-	if err != nil {
-		sip.MoveToDraining(err)
-		return nil, sip.DrainHelper()
-	}
-
-	if progressUpdate != nil {
-		progressBytes, err := protoutil.Marshal(progressUpdate)
+	select {
+	case progressUpdate, ok := <-sip.checkpointCh:
+		if ok {
+			progressBytes, err := protoutil.Marshal(progressUpdate)
+			if err != nil {
+				sip.MoveToDraining(err)
+				return nil, sip.DrainHelper()
+			}
+			row := rowenc.EncDatumRow{
+				rowenc.DatumToEncDatum(types.Bytes, tree.NewDBytes(tree.DBytes(progressBytes))),
+			}
+			return row, nil
+		}
+	case err := <-sip.errCh:
 		if err != nil {
```

Comment:
I think err cannot be nil here, right? if it can then we will drop the second error which is not great..

Reply:
It cannot. I removed this and moved an `if err == nil` into sendError just as a bit of precaution.